docs/news.rst

```diff
@@ -3,6 +3,148 @@
 Release notes
 =============
 
+Scrapy 1.4.0 (2017-XX-XX)
+-------------------------
+
+Scrapy 1.4 does not bring that many breathtaking new features
+but quite a few handy improvements nonetheless.
+
+Scrapy now supports anonymous FTP sessions with customizable user and
+password via the new :setting:`FTP_USER` and :setting:`FTP_PASSWORD` settings.
+**And if you're using Twisted version 17.1.0 or above, FTP is now available
+with Python 3.**
+
+Link extractors now work similarly to what a regular modern browser would
+do. Especially, leading and trailing whitespace are removed from attributes
+(think ``href="   http://example.com"``) when building ``Link`` objects.
+This whitespace-stripping also happens for ``action`` attributes with
+``FormRequest``.
+**Please also note that link extractors do not canonicalize URLs by default
+anymore.** This was puzzling users every now and then, and it's not what
+browsers do in fact, so we removed that extra transformation on extractred
+links.
+
+There's a new ``response.follow()`` shortcut for creating URLs directly
+from the response instance in callbacks.
+For example, instead of::
+
+    scrapy.Request(response.urljoin(somehrefvalue))
+
+you can now use the simpler::
+
+    response.follow(somehrefvalue)
```

Comment:
Does it return a `Request` object? The description above may lead people to think that it just creates the URL from the `Response` instance (which is what `response.urljoin` does).

What about:
> There's a new ``response.follow()`` shortcut for creating requests directly from a response instance and a relative URL.